zqd/space/space.go

```diff
@@ -129,23 +129,27 @@ func (s Space) PcapSearch(req api.PacketSearch) (*pcap.SearchReader, io.Closer,
 	case "icmp":
 		search = pcap.NewICMPSearch(req.Span, req.SrcHost, req.DstHost)
 	default:
-		return nil, nil, fmt.Errorf("unsupported proto type: %s", req.Proto)
+		return nil, "", fmt.Errorf("unsupported proto type: %s", req.Proto)
 	}
 	f, err := os.Open(s.PacketPath())
 	if err != nil {
-		return nil, nil, err
+		return nil, "", err
 	}
 	slicer, err := pcap.NewSlicer(f, index, req.Span)
 	if err != nil {
 		f.Close()
-		return nil, nil, err
+		return nil, "", err
 	}
 	pcapReader, err := pcapio.NewReader(slicer)
 	if err != nil {
 		f.Close()
-		return nil, nil, err
+		return nil, "", err
 	}
-	return search.Reader(pcapReader), f, nil
+	r := search.Reader(pcapReader)
+	return struct {
+		io.Reader
+		io.Closer
+	}{r, f}, search.ID(), nil
```

Comment:
I like this approach, would suggest you have it return a 'SearchReadCloser' or something that looks like:
```go
type SearchReadCloser {
  *pcap.SearchReader
   f *os.File
}

func (c *SearchReadCloser) Close() error {
  return c.f.Close()
}
```

With above change, you'll be able to pass just this instance and an error; the code in handlers.go can call `.ID()` on it directly.

Reply:
sure